src/internal/Observable.ts

```diff
@@ -307,28 +358,57 @@ export class Observable<T> implements Subscribable<T> {
    * polyfill Promise, or you create an adapter to convert the returned native promise
    * to whatever promise implementation you wanted.
    */
-  forEach(next: (value: T) => void, promiseCtor: PromiseConstructorLike): Promise<void>;
+  forEach(nextHandler: (value: T) => void, promiseCtor: PromiseConstructorLike): Promise<void>;
+
+  forEach(nextHandler: (value: T) => void, promiseCtorOrSignal?: PromiseConstructorLike | AbortSignal): Promise<void> {
+    let promiseCtor: PromiseConstructorLike | undefined;
+    let signal: AbortSignal | undefined;
+    if (isAbortSignal(promiseCtorOrSignal)) {
+      signal = promiseCtorOrSignal;
+    } else {
+      promiseCtor = promiseCtorOrSignal;
+    }
 
-  forEach(next: (value: T) => void, promiseCtor?: PromiseConstructorLike): Promise<void> {
     promiseCtor = getPromiseCtor(promiseCtor);
 
     return new promiseCtor<void>((resolve, reject) => {
+      if (signal) {
+        signal.addEventListener(
+          'abort',
+          () => {
+            // TODO: Chrome and Firefox both use DOMException here for now
+            // This should probably be something better defined. MDN documentation
+            // says there is supposed to be an AbortError, but I haven't found any
+            // implementations in any runtimes. For now, we can document this
+            // as the users needing to check `err.name === 'AbortError'` and
+            // adjust over time.
+            const err = new Error('Abort exception');
+            err.name = 'AbortError';
+            reject(err);
+          },
+          { once: true }
+        );
```

Comment:
Maybe I'm missing something, but doesn't this need cleanup? `once: true` will clean the event listener if it fires, but it seems that if the observable naturally completes, then this would keep listening to the AbortSignal, as there's nothing removing it.

Reply:
You're absolutely right! Thank you for catching that!